src/network/wait-online/link.c

```diff
@@ -120,6 +120,8 @@ int link_update_rtnl(Link *l, sd_netlink_message *m) {
 int link_update_monitor(Link *l) {
         assert(l);
 
+        l->required_for_online = sd_network_link_get_required_for_online(l->ifindex);
```

Comment:
the function returns an "int", which might very well be negative due to errors. The variable takes a bool however.

Now it's fine ignoring errors here, but this should be made explicit, for example by doing `l->r_f_o = s_n_l_g_r_f_o(l->ifindex) != 0`. That way it is clear that you are aware of the difference of types, and that errors should be treated as true